src/Http/TraceContextAwareHttpClient.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DR\SymfonyRequestId\Http;
+
+use DR\SymfonyRequestId\Service\TraceContextService;
+use DR\SymfonyRequestId\TraceContext;
+use DR\SymfonyRequestId\TraceId;
+use DR\SymfonyRequestId\TraceStorageInterface;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+class TraceContextAwareHttpClient implements HttpClientInterface, ResetInterface, LoggerAwareInterface
+{
+    public function __construct(
+        private HttpClientInterface            $client,
+        private readonly TraceStorageInterface $storage,
+        private readonly TraceContextService   $traceContextService
+    ) {
+    }
+
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        if ($this->storage->getTrace() instanceof TraceContext) {
```

Comment:
Instead of having specific TraceContextService input here. Pass the `method`, `url`, and `options` to the service and let the service manipulate the options. This way we can swap in a different service and set different headers.

Suggestion:
`$options = $this->traceContextService->applyTraceHeaders(string $method, string $url, array $options): array`

Or something like

`$options['headers'] += $this->traceContextService->getTraceheaders(string $method, string $url, array $options): array`